### PR TITLE
fix: block private/loopback addresses in fetch_remote_image (#275)

### LINKED ISSUE #275: [Tech Debt] Critical - Admin SSRF: fetch_remote_image does not block private/loopback IP ranges
`db/src/author_photos.rs:285` guards `fetch_remote_image` with a scheme check only — it rejects non-`http`/`https` URLs but does not validate the resolved IP address. An admin can supply `http://127.0.0.1:6379/`, `http://169.254.169.254/latest/meta-data/` (AWS IMDSv1), `http://192.168.1.1/`, or any other RFC 1918 / loopback / link-local / unique-local address. The server will resolve the hostname, open a TCP connection, stream the response, check `Content-Type: image/*`, and either store or reject the bytes. The TCP connection and the initial fetch still happen unconditionally — even a rejected MIME type has already proved the port open.

This is **Critical** severity. The SSRF is admin-only (`AdminUser` extractor at the calling handler), which limits exposure to compromised admin accounts. However in the self-hosted use case described in the roadmap, the operator is typically the sole user and sole admin, so a compromised session is effectively full-system compromise. The cloud metadata IMDS path (`169.254.169.254`) is reachable from any cloud VM and returns IAM credentials in plaintext.

Sources: `db/src/author_photos.rs:284–295` — `fetch_remote_image` scheme guard and immediate `reqwest::Client::get(url).send()`. `server/src/backend/author_photos.rs` — the calling `PUT /api/authors/:id/photo/url` handler that wraps this function with `AdminUser` extractor.

Before issuing the `reqwest` request, resolve the target hostname to an `IpAddr` and reject any address that is loopback (`is_loopback()`), private RFC 1918 / ULA (`is_private()` on `Ipv4Addr`, `is_unicast_site_local()` / `is_unique_local()` on `Ipv6Addr`), link-local (`is_link_local()`), or multicast. The `hickory-resolver` or `std::net::ToSocketAddrs` can be used for pre-resolution. Alternatively, use a custom `reqwest` connector that wraps the TCP connect and refuses connections to blocked ranges at the socket layer, which also handles DNS rebinding attacks.

### PR DESCRIPTION
**Pull request by seamus-sloan**
## Summary
- `db::author_photos::fetch_remote_image` only validated the URL scheme. An admin could paste `http://127.0.0.1:6379/`, `http://169.254.169.254/latest/meta-data/` (AWS IMDSv1), `http://192.168.1.1/`, etc., and the server would open the TCP connect before any content-type / size check — a critical SSRF (#275). On a cloud VM this exposes IAM credentials in plaintext on any compromised admin session.
- Add a `RemoteImageConfig` SSRF guard: parse the URL, resolve via `tokio::net::lookup_host`, reject any resolved IP that is loopback / unspecified / RFC1918 / link-local (incl. AWS IMDS at 169.254.169.254) / multicast / broadcast / documentation / CGN / benchmarking / IPv6 ULA (fc00::/7) / IPv6 link-local (fe80::/10) / IPv6 documentation / IPv6-mapped IPv4 that falls into any of the above. Then build a `reqwest::Client` pinned to the validated `SocketAddr` set via `.resolve_to_addrs(...)` so DNS rebinding between our lookup and reqwest's own resolution can't sneak in a private IP.
- IP-range bit-checks are inlined because `Ipv4Addr::is_shared`, `Ipv6Addr::is_unique_local`, and `Ipv6Addr::is_unicast_link_local` are nightly-only on stable.
- Production callers (`put_author_photo_url`, `rpc_set_author_photo_url`) go through the default strict config. The `allow_private_addresses` escape hatch lives on `AppState` behind `#[cfg(test)]` so existing wiremock-backed integration tests can keep hitting `127.0.0.1` without softening the production path.

## Test plan
- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --all-targets -- -D warnings` (project default; `--all-features` is incompatible with the `web`/`mobile` feature split and breaks on unrelated code on `main`)
- [x] `cargo test -p omnibus-db` — 349 tests pass, including 17 new `is_blocked_address` / `fetch_remote_image_with` unit tests covering every blocked IP class, a positive case per IP family, the IPv6-mapped loopback edge case, and the test-config escape hatch
- [x] `cargo test -p omnibus` — 129 tests pass, including the new `api_put_author_photo_url_blocks_private_ip_targets` integration test that drives the production `AppState` against eight SSRF URLs (loopback, IMDS, RFC1918, IPv6 loopback, IPv6-mapped loopback) and asserts each returns 400
- [x] `cargo test -p omnibus-frontend --features server` — 100 tests pass

## Notes
- One follow-up worth considering: the `rpc_set_author_photo_url` RPC handler still calls `fetch_remote_image` (the strict default) rather than threading a config through. That's fine today because RPC and REST agree on strict-only, but if we ever want the same loopback escape hatch on the RPC side for tests, the cleaner shape would be a small struct shared between `AppState` and the RPC `PoolExt` context. Out of scope for this fix.
- The fix builds a fresh `reqwest::Client` per strict-mode call (so it can pin `.resolve_to_addrs(...)`), forgoing the shared connection pool for this code path. Author-photo URL uploads are an interactive admin action firing at most a handful of times per day, so the per-request TLS handshake cost is irrelevant. The `OpenLibraryConfig` cascade and the lax test path keep using the shared client.
- `CLAUDE.md` updated to document the new `RemoteImageConfig` SSRF guard alongside the existing `OpenLibraryConfig` note for `db/src/author_photos.rs`.

Closes #275

https://claude.ai/code/session_01Jr6c66demnTfNf1XYGTtwX

---
_Generated by [Claude Code](https://claude.ai/code/session_01Jr6c66demnTfNf1XYGTtwX)_

### CODE REVIEW

File: db/src/author_photos.rs [hunk 1]
```diff
@@ -281,18 +424,47 @@ pub enum FetchRemoteImageError {
...
+    // grep-auditable, and the production handlers never set it.
+    let client = if config.allow_private_addresses {
+        shared_client()?
+    } else {
+        let (host, addrs) = validated_resolve(url).await?;
+        reqwest::Client::builder()
+            .user_agent(default_user_agent())
+            .resolve_to_addrs(&host, &addrs)
+            .build()?
+    };
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply db/src/author_photos.rs]: Addressed in c6f0813 — the strict-mode reqwest client is now built with `.redirect(reqwest::redirect::Policy::none())` (author_photos.rs:464), so a 302 surfaces as `BadStatus(302)` instead of being silently chased to its target. Added `fetch_remote_image_does_not_follow_redirects_to_private_ips` (author_photos.rs:917) which stands up a wiremock server returning a 302 with `Location: http://127.0.0.1:1/` and asserts strict-mode `fetch_remote_image_with` rejects the redirect rather than re-resolving the new host.

---
_Generated by [Claude Code](https://claude.ai/code)_
